src/main/java/me/lucien/minesweeper/domain/RoomManager.java

```diff
@@ -0,0 +1,69 @@
+package me.lucien.minesweeper.domain;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+@Component
+public class RoomManager {
+
+    @Value("${room.capacity}")
+    private int capacity;
+
+    @Value("${room.delay}")
+    private int delay;
+
+    private Map<Integer, Room> roomMap = new HashMap<>();
+    private Map<Integer, ScheduledFuture<?>> taskMap = new HashMap<>();
+    private ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor();
+
+    public Room createRoom(int width, int height) {
+        if (roomMap.size() >= capacity) {
+            return null;
+        }
+
+        Room room = new Room(width, height);
+        int id = room.getId();
+        roomMap.put(id, room);
```

Comment:
When there are multiple concurrent requests, roomMap isn't thread-safe and there will be numerous race conditions. Use ConcurrentHashMap instead of HashMap. Or make methods synchronized.